src/codegen/compile.ml

```diff
@@ -5381,6 +5350,177 @@ module StableMem = struct
 
 end (* StableMemory *)
```

Comment:
`StableMem` vs `StableMemory` is a bit confusing for people that haven't seen all of the stages of evolution.

Is there a plan to consolidate these in some future PR, I wonder?

Reply:
Yeah, I agree. I wanted to keep the StableMem name for the old module which is largely unchanged. 

Maybe StableMemoryAdapter (since it adapts between old an new?), or StableMemoryPrims?

Reply:
StableMem basically enforces --max-stable-pages and hides from the user any IC stable memory added for stable variables.

StableMemory provides the guts of the StableMemoryPrims, dynamically switching between legacy and regions.